vassal-app/src/main/java/VASSAL/tools/ProblemDialog.java

```diff
@@ -353,27 +338,55 @@ public void run() {
 
     if (thrown != null) logger.error("", thrown);
 
-    return DialogUtils.enqueue(new Runnable() {
-      @Override
-      public void run() {
-        DetailsDialog.showDialog(
-          parent,
-          title,
-          heading,
-          message,
-          details,
-          Resources.getString("Dialogs.disable"),
-          Resources.getString("Dialogs.show_details"),
-          Resources.getString("Dialogs.hide_details"),
-          messageType,
-          key
-        );
-      }
-    });
+    return DialogUtils.enqueue(() -> DetailsDialog.showDialog(
+      parent,
+      title,
+      heading,
+      message,
+      details,
+      Resources.getString("Dialogs.disable"),
+      Resources.getString("Dialogs.show_details"),
+      Resources.getString("Dialogs.hide_details"),
+      messageType,
+      key
+    ));
   }
 
   private static Frame getFrame() {
     return GameModule.getGameModule() == null
       ? null : GameModule.getGameModule().getPlayerWindow();
   }
+
+  public static Future<?> showDeprecated(String date) {
+    final StackWalker.StackFrame frame =
+      StackWalker.getInstance(Set.of(StackWalker.Option.RETAIN_CLASS_REFERENCE), 2)
+        .walk(f -> f.skip(1).limit(1).collect(Collectors.toList())).get(0);
+    final String method = frame.getClassName() + "." + frame.getMethodName();
+    String expiry;
+    try {
+      expiry = LocalDate.parse(date, DateTimeFormatter.ofPattern("uuuu-M-d")).plusYears(1).format(DateTimeFormatter.ofPattern("d-MMM-uuuu", Resources.getLocale()));
```

Comment:
Is there a reason we're not using the `ISO_LOCAL_DATE` formatter in both places here?

Reply:
On Input, ISO_LOCAL_DATE is inflexible on the number of digits used for year, month or day. "uuuu-M-D" will happily parse "20-8-4" or "2020-08-04".

On output, I find the '3 Aug 2020' form far friendlier for multi-locale usage. No change of confusion. That's just my opinion though.

Reply:
That's exactly why I like ISO-8601 dates. :)

I think we should use a yyyy-mm-dd format for the deprecation annotation. Showing month names in messages is ok.